src/main/java/calculator/StringExpression.java

```diff
@@ -0,0 +1,54 @@
+package calculator;
+
+public class StringExpression {
+
+  private String totalExpression;
+
+  public StringExpression(String totalExpression) {
+    ExpressionValidator.validate(totalExpression);
+    this.totalExpression = totalExpression;
+  }
+
+  private class ThreeExpression {
```

Comment:
멤버 클래스에 static 키워드가 없으면 메모리 누수의 원인이 될 수도 있으니 가능한 static 키워드를 붙여주는게 좋습니다. 😄 
```suggestion
  private static class ThreeExpression {
```
자세한건 내용은 아래 링크를 참고해주세요.

[이펙티브 자바 3판 아이템24](https://www.google.com/search?newwindow=1&sxsrf=ALeKk01UUYIdxjbZU5NG0IUy0sJLFbx6zg%3A1614814734863&ei=Dh5AYIadNNCsmAWFnaOwCw&q=%EC%9D%B4%ED%8E%99%ED%8B%B0%EB%B8%8C+%EC%9E%90%EB%B0%94+%EC%95%84%EC%9D%B4%ED%85%9C+24&oq=%EC%9D%B4%ED%8E%99%ED%8B%B0%EB%B8%8C+%EC%9E%90%EB%B0%94+%EC%95%84%EC%9D%B4%ED%85%9C+24&gs_lcp=Cgdnd3Mtd2l6EAMyBAgAEB46BAgjECc6AggAOgcIABCHAhAUUJqEB1iOpAdg96QHaAdwAHgCgAGzAYgByxaSAQQwLjIzmAEAoAEBqgEHZ3dzLXdpesABAQ&sclient=gws-wiz&ved=0ahUKEwiG3JjVpZXvAhVQFqYKHYXOCLYQ4dUDCA0&uact=5)